build-scripts/build_llvm.py

```diff
@@ -102,12 +102,27 @@ def build_llvm(llvm_dir, platform, backends, projects, use_clang=False, extra_fl
         "default": [],
     }
 
+    normal_backends = [s for s in backends if s != "ARC" and s != "Xtensa"]
+    experimental_backends = [s for s in backends if s == "ARC" or s == "Xtensa"]
+
     LLVM_TARGETS_TO_BUILD = [
-        '-DLLVM_TARGETS_TO_BUILD:STRING="' + ";".join(backends) + '"'
-        if backends
+        '-DLLVM_TARGETS_TO_BUILD:STRING="' + ";".join(normal_backends) + '"'
+        if normal_backends
         else '-DLLVM_TARGETS_TO_BUILD:STRING="AArch64;ARM;Mips;RISCV;X86"'
     ]
 
+    for experimental_backend in experimental_backends:
+        # if not on ARC platform, but want to add expeirmental backend ARC as target
+        if platform != "ARC" and experimental_backend == "ARC":
+            LLVM_TARGETS_TO_BUILD.extend(
+                LLVM_EXTRA_COMPILE_OPTIONS[experimental_backend.lower()]
+            )
+        elif platform != "Xtensa" and experimental_backend == "Xtensa":
+            print(
+                "Currently it's not supported to build Xtensa backend on non-Xtensa platform"
+            )
+            return None
+
```

Comment:
seems the `for` loop is not necessary. Only need to handle two cases, `platform != "ARC" and experimental_backend == "ARC"` and `platform != "Xtensa" and experimental_backend == "Xtensa"`, others can be ignored.